### PR TITLE
Feature: support for new XML format (per SonarQube)

### PR DESCRIPTION
**Pull request by jpaulin**
- Feature addition: XML schema support as per new SonarQube
- adds a corresponding flag support in karma.conf.js for the new XML mode
- proposing "xmlVersion" field into karma.conf.js section `junitReporter`
  - 0 or omitted: old legacy behavior (such won't break happy users)
  - 1: uses new XML version output (makes SonarQube 6.2 and newer users happy)
  - in the future, even newer XML versions can be numbered 2,3,4, and so on

- related switch added to karma.conf.js: xmlVersion (if omitted => works in legacy mode, if defined -> works in new XML mode)

- so, commit code keeps the current functioning of karma-junit-reporter as-is,
  nothing changed. Unless user changes the karma.conf.js configuration file

## Validation of new XML report

The XSD schema ("specs") used for testing is also included in the commit
(file sonar-unit-tests.xsd)

New XML output report was validated against the specs with 'xmllint' tool. If you want to replicate XML validation, install the libxml2 and libxml2-utils on Linux. See eg. http://xmlsoft.org/ for references.

### CODE REVIEW

**Comment by jpaulin** [comment]
If someone wants to test this, would be great. Don't test in production, necessarily. 

To test:
drop this new index.js into your project-local karma-junit-reporter. For example, I have now:
./node_modules/karma-junit-reporter/index.js

So 'cp' (copy) this new index.js over that old mentioned above.

Further: 

1. To test that nothing was broken in old code:
just run your karma -related testing as you used to

2. To test the **new** thing, ie. XML format, edit your karma.conf.js file and insert **xmlVersion: 1**
into the junitReporter -section

Thanks,
J

**Comment by tubbynl** [comment]
thank you! it would be really cool if junit support on sonar is back :) +1

File: package.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
     "index.js"
   ],
   "scripts": {
-    "test": "grunt"
+    "jp-test": "mocha --compilers coffee:coffee-script"
```
Comment (dignifiedquire) [inline package.json]: why this change?

File: sonar-unit-tests.xsd [hunk 2]
```diff
@@ -0,0 +1,54 @@
+<?xml version="1.0"?>
+<xs:schema xmlns:xs="http://www.w3.org/2001/XMLSchema">
```
Comment (dignifiedquire) [inline sonar-unit-tests.xsd]: what's the purpose of this file?

Comment (dignifiedquire) [inline sonar-unit-tests.xsd]: I just saw you mention it is used for testing, but I don't see any tests added for this

File: package.json (hunk 1 above)
Comment (jpaulin) [inline package.json]: Good you noticed, this was a leftover from some development stage. No need to change test command, should be just as it was.

File: sonar-unit-tests.xsd (hunk 2 above)
Comment (jpaulin) [inline sonar-unit-tests.xsd]: Yes, the .xsd file was included to validate the new XML file against. So in my case I used a external well-known xml tool (comes with the libxml-utils package in Linux) to do the actual validation. Can be dropped, certainly, doesn't affect the code.

Comment (dignifiedquire) [inline sonar-unit-tests.xsd]: Would be great if you could add a test to the current tests that generates the file and validates it, pretty sure there is an npm module to do that :)

Comment (jpaulin) [inline sonar-unit-tests.xsd]: I like the idea. What is a bit of a thing, is finding a suitably "nice" xml validator that wouldn't cause trouble in Travis. Is it okay if the xml package requires node-gyp during install? Seems quite many ride on top of something of native code.

Comment (dignifiedquire) [inline sonar-unit-tests.xsd]: as it's only used in testing, I don't mind

Comment (jpaulin) [inline sonar-unit-tests.xsd]: Seems to choke at node-gyp with Travis. I had bit of omen about this.. Basically I might check out alternatives since now most of the things are already in place. Unless someone knows how to get the node-gyp kicking well.

Comment (jpaulin) [inline sonar-unit-tests.xsd]: Solved, now passes in Travis. The node-gyp compilation started working, when Travis instructed to use g++4.8 as the C compiler suite (CXX).

**Comment by jpaulin** [comment]
All go from this side. If you care to review this PR further, would be great.

**Comment by dignifiedquire** [comment]
Thank you :octocat:
